[M/voucher-44, 54] 바우처 삭제 및 조회 기능 추가

KwonYeKyeong opened the pull request:
- 기능 구현이 우선일꺼 같아서 지난 피드백은 추후에 리팩토링 하겠습니다!!

KwonYeKyeong committed:
[M] refactor : 예외처리 형식에 맞게 수정 (#41)

KwonYeKyeong committed:
[M] feat : 바우처를 삭제할 수 있다 - service, controller (#44)

KwonYeKyeong committed:
[M] chore : JSON으로 변환 시 snake case로 변경

KwonYeKyeong committed:
[M] feat : 바우처를 조회할 수 있다 - service, controller (#54)

KwonYeKyeong committed:
[M] test : 바우처 조회 및 삭제 service 단위 테스트, 통합 테스트 작성 (#44, #54)

KwonYeKyeong committed:
[M] test : voucher 통합 테스트 수정

seung-hun-h commented on ourhome-market/src/main/java/com/armand/ourhome/market/voucher/service/VoucherService.java:
```diff
@@ -49,11 +56,17 @@ public VoucherDto update(Long id, RequestVoucher request) {
 
     if (byId.isEmpty()) {
-      throw new EntityNotFoundException(MESSAGE_VOUCHER_NOT_FOUND);
+      throw new VoucherNotFoundException(MESSAGE_VOUCHER_NOT_FOUND);
```
id도 함께 예외 메시지에 전달하면 좋을 것 같아요

seung-hun-h commented on ourhome-market/src/main/java/com/armand/ourhome/market/voucher/service/VoucherService.java:
```diff
@@ -49,11 +56,17 @@ public VoucherDto update(Long id, RequestVoucher request) {
+      throw new VoucherNotFoundException(MESSAGE_VOUCHER_NOT_FOUND);
     }
 
     return byId.get();
```
validation을 담당하는 메서드가 엔티티를 반환하는 것은 어색하네요

seung-hun-h commented on ourhome-market/src/main/java/com/armand/ourhome/market/voucher/service/VoucherService.java:
```diff
@@ -63,16 +76,15 @@ private void validateDuplicateVoucher(RequestVoucher request) {
 
   private void validateVoucherType(RequestVoucher request, Voucher voucher) {
     boolean differentType = request.getVoucherType().isDifferentType(voucher);
 
     if (differentType) {
-      throw new DifferentTypeVoucherException(MESSAGE_VOUCHER_TYPE_NOT_SAME,
-          ErrorCode.INVALID_TYPE_VALUE);
+      throw new DifferentTypeVoucherException(MESSAGE_VOUCHER_TYPE_NOT_SAME);
```
voucherType validation은 VoucherType의 책임이 아닐까 생각합니다

seung-hun-h reviewed (commented):
리뷰 남겼습니다 확인해주세요!

KwonYeKyeong committed:
[M] refactor : 피드백 반영 (#59)

- 예외메세지 전달 시 id값을 함께 전달하도록 수정
- validation 메서드의 반환값 수정
- VoucherType class에서 type 중복검사를 하도록 수정
